platform/viewer/src/connectedComponents/ViewerRetrieveStudyData.js

```diff
@@ -1,77 +1,118 @@
-import React, { Component } from "react";
-import { metadata, studies, utils } from "@ohif/core";
+import React, { Component } from 'react';
+import { metadata, studies, utils, log } from '@ohif/core';
 
-import ConnectedViewer from "./ConnectedViewer.js";
-import PropTypes from "prop-types";
-import { extensionManager } from "./../App.js";
+import ConnectedViewer from './ConnectedViewer.js';
+import PropTypes from 'prop-types';
+import { extensionManager } from './../App.js';
 
-const { OHIFStudyMetadata } = metadata;
+const { OHIFStudyMetadata, OHIFSeriesMetadata } = metadata;
 const { retrieveStudiesMetadata } = studies;
 const { studyMetadataManager, updateMetaDataManager } = utils;
 
 class ViewerRetrieveStudyData extends Component {
   static propTypes = {
     studyInstanceUids: PropTypes.array.isRequired,
     seriesInstanceUids: PropTypes.array,
-    server: PropTypes.object
+    server: PropTypes.object,
   };
 
-  state = {
-    studies: null,
-    error: null
-  };
-
-  async componentDidMount() {
-    // TODO: Avoid using timepoints here
-    //const params = { studyInstanceUids, seriesInstanceUids, timepointId, timepointsFilter={} };
-    const { studyInstanceUids, seriesInstanceUids, server } = this.props;
-
-    try {
-      const studies = await retrieveStudiesMetadata(
-        server,
-        studyInstanceUids,
-        seriesInstanceUids
-      );
+  constructor(props) {
+    super(props);
+    this.isReady = false;
+    this.state = {
+      studies: null,
+      error: null,
+    };
+  }
 
-      // Render the viewer when the data is ready
-      // TODO: CLEAR THIS SOMEWHERE ELSE
-      studyMetadataManager.purge();
+  triggerDataLoad() {
+    const { props } = this;
+    retrieveStudiesMetadata(
+      props.server,
+      props.studyInstanceUids,
+      props.seriesInstanceUids
+    ).then(
+      studies => {
+        if (this.isReady) {
+          this.setStudies(studies);
+        }
+      },
+      error => {
+        if (this.isReady) {
+          this.setState({ error: true });
+        }
+        log.error(error);
+      }
+    );
+  }
 
+  setStudies(givenStudies) {
+    if (Array.isArray(givenStudies) && givenStudies.length > 0) {
+      const sopClassHandlerModules =
+        extensionManager.modules['sopClassHandlerModule'];
       // Map studies to new format, update metadata manager?
-      const updatedStudies = studies.map(study => {
+      const studies = givenStudies.map(study => {
         const studyMetadata = new OHIFStudyMetadata(
           study,
           study.studyInstanceUid
```

Comment:
I know you didn't write this, but I don't understand why we need to pass `studyInstanceUid` when it already exists on `study`